storage/perfschema/CMakeLists.txt

```diff
@@ -280,6 +280,11 @@ table_replication_applier_status_by_worker.cc
 #table_replication_group_member_stats.cc
 )
 
+# These may warning on warning: implicit declaration of function
+# which isn't good enough as a compile test.
+set(SAVED_CMAKE_REQUIRED_FLAGS "${CMAKE_REQUIRED_FLAGS}")
```

Comment:
Nah, this is proprietary, an the amount of warnins depends on the version of GCC. Better, CHECK_CXX_SOURCE_COMPILES ,  i.e C++ instead of C, there are no "implicit declaration of a function" in C++.
Also, a link error should by caught by CHECK_C_SOURCE_COMPILES  (From the documentation "Check once that the source supplied in <code> can be built, linked as an executable, and then run. The <code> must contain at least a main() function") ,but perhaps AIX linker returns 0 even when it fails, I remember some irregularities.

Reply:
There  is probably something about this funny linker option 
storage/perfschema/CMakeLists.txt:  
```cmake
SET(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,-berok")
```
that allows linker to return 0 if there are unresolved symbols :)
We can blame cmake, and try_compile, but if we use this option, all bets are off

Reply:
I suggest removing these "beroks", and either do not build the test executables (there is really not much value in them)
or, if absolutely necessary, to add
``cmake
if(AIX)
  target_link_options(target PRIVATE "-Wl,-berok") for targets that fail.
endif()
```
This is preferable to changing CMAKE_EXE_LINKER_FLAGS, and has less side-effect

Though, personally, I'd rather *not* workaround linker bugs, and build unimportant unit-and-other-test executables with if(NOT AIX)